cmake/legate_helper_functions.cmake

```diff
@@ -0,0 +1,384 @@
+macro(legate_include_rapids)
+  if (NOT _LEGATE_HAS_RAPIDS)
+    if(NOT EXISTS ${CMAKE_BINARY_DIR}/RAPIDS.cmake)
+      file(DOWNLOAD https://raw.githubusercontent.com/rapidsai/rapids-cmake/branch-22.08/RAPIDS.cmake
+           ${CMAKE_BINARY_DIR}/RAPIDS.cmake)
+    endif()
+    include(${CMAKE_BINARY_DIR}/RAPIDS.cmake)
+    include(rapids-cmake)
+    include(rapids-cpm)
+    include(rapids-cuda)
+    include(rapids-export)
+    include(rapids-find)
+    set(_LEGATE_HAS_RAPIDS ON)
+  endif()
+endmacro()
+
+function(legate_default_cpp_install target)
+  set(options)
+  set(one_value_args EXPORT)
+  set(multi_value_args)
+  cmake_parse_arguments(
+    OPT
+    "${options}"
+    "${one_value_args}"
+    "${multi_value_args}"
+    ${ARGN}
+  )
+
+  if (NOT OPT_EXPORT)
+    message(FATAL_ERROR "Need EXPORT name for legate_default_install")
+  endif()
+
+  legate_include_rapids()
+
+  rapids_cmake_install_lib_dir(lib_dir)
+
+  install(TARGETS ${target}
+          DESTINATION ${lib_dir}
+	  EXPORT ${OPT_EXPORT})
+
+  set(code_string "set(legate_core_DIR ${legate_core_DIR})")
```

Comment:
Do you want to generate relocatable packages? This will bind this package to where-ever legate core is on the local machine.

Reply:
Yeah, I was trying to be too clever in the downstream app. I removed this.